department-of-veterans-affairs/vets-api
# API-37730-tracked-items-5103

## Summary

- Adds validation for schema. 
- Adds 5103 example request body. 
- Adds dev docs for optional request body. 
- Passes param of trackedItems to builder job, which uploads them to BD. 
- Adds request tests. 
- Adds variable to rswag for optional request body.

## Related issue(s)

- [API-37730](https://jira.devops.va.gov/browse/API-37730)

## Testing done


*** Testing Instructions***
1. In VBMS, create a tracked Item (can be anything) for the claim you plan to test
2. Use our claims show to get the tracked item id
![Screenshot 2024-07-17 at 12 25 21 PM](https://github.com/user-attachments/assets/ecae0e04-ceaa-478f-9b3d-6b6e8fc10b39)

3. Use the tracked item ID to submit a 5103 using the claim id, & tracked item id in the request body
 - Postman with and without request body: {{uri}}/services/claims/v2/veterans/1012666073V986297/claims/600527334/5103
```
{
    "data": {
        "type": "form/5103",
        "attributes": {
            "trackedItems": [-insert-from-steps-above]
        }
    }
}
```
4. Use the claim show to check for a supporting doc with the tracked item id that you submitted 
![Screenshot 2024-07-17 at 12 24 33 PM](https://github.com/user-attachments/assets/54c207e4-cf78-4b7b-a4d6-6a07f4ccced6)

## If the claim has been completed or canceled, you will get a 400 from the evss docker container************

## Screenshots
![Screenshot 2024-07-17 at 10 13 00 AM](https://github.com/user-attachments/assets/b3f1c934-eff7-4930-82d2-972fc36ac88d)
![Screenshot 2024-07-17 at 10 11 53 AM](https://github.com/user-attachments/assets/b688e224-9674-468a-805f-7dbae6e50c8b)

## What areas of the site does it impact?
	modified:   modules/claims_api/app/controllers/claims_api/v2/veterans/evidence_waiver_controller.rb
	modified:   modules/claims_api/app/sidekiq/claims_api/evidence_waiver_builder_job.rb
	modified:   modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json
	new file:   modules/claims_api/config/schemas/v2/5103.json
	modified:   modules/claims_api/lib/bd/bd.rb
	new file:   modules/claims_api/lib/claims_api/v2/params_validation/evidence_waiver_submission.rb
	new file:   modules/claims_api/lib/claims_api/v2/params_validation/evidence_waiver_submission/evidence_waiver_submission_info.rb
	new file:   modules/claims_api/lib/claims_api/v2/params_validation/evidence_waiver_submission/evidence_waiver_submission_validator.rb
	new file:   modules/claims_api/lib/claims_api/v2/params_validation/evidence_waiver_submission/main.rb
	new file:   modules/claims_api/spec/fixtures/v2/veterans/5103/form_5103_api.json
	modified:   modules/claims_api/spec/requests/v2/veterans/evidence_waiver_request_spec.rb
	modified:   modules/claims_api/spec/requests/v2/veterans/rswag_evidence_waiver_spec.rb
	modified:   modules/claims_api/spec/support/swagger_shared_components/v2.rb


## Acceptance criteria
